### DISCUSSION TITLE
How to save large datasets into hdf5?

### DISCUSSION THREAD

**Discussion by Spooyler** [hyperspy/hyperspy#3452] [Q&A]
I have an issue, that I have relatively large data files in RPL/RAW format that are generated on a 32 bit windows 7 machine. Files are around 500-1000 MB. Moving these files is a pain, so I prefer to convert them first:
```
import hyperspy.api as hs

s = hs.load(*.rpl).as_signal1D(0, optimize=False)
s.change_dtype('float_32')#they are initially int32 but displayed weird if not converted;omitting this step does nothing
s.save('something.hdf5', compression='lzf')
```
This is just an MRE, but I found other parts have no real effect.
Basically I get 10 MB files which are much easier to move.

The problem is I can't do this on the source machine as I quickly run out of memory. If I use `as_lazy()` I still run out of memory during export.

Anyone have similar experiences or managed to solve similar issues?

EDIT: I should add, if I manage to do the conversion, loading it into hyperspy is not an issue anymore. so:
`s = hs.load('something.hdf5')` works without any memory issues!

**Comment by ericpre** [comment]
Have you tried to load lazily?
`as_lazy` converts to lazy signal but the data may be loaded to memory before that - this may incorrect with the rpl/raw reader, as it may be using memory mapping.

```python
import hyperspy.api as hs

s = hs.load(*.rpl, lazy=True)
s.save("fname.hspy")
```

**Reply by Spooyler** [reply]
I tried lazy loading yes. As you said, the rpl reader uses memory mapping, apparently when the data is exported it will load the whole thing into memory.

**Comment by Spooyler** [comment]
Ended up moving the whole thing to a 64 bit system, and created a local network for file sharing. 
32 bit just doesn't worth the effort.
